test/e2e/e2e_test.go

```diff
@@ -98,86 +100,119 @@ func RcByNameContainer(name string, replicas int32, labels map[string]string, gr
-	if err != nil || len(evictionPolicyGroupVersion) == 0 {
-		klog.Fatalf("%v", err)
-	}
-	stopChannel := make(chan struct{})
-	nodes, err := nodeutil.ReadyNodes(ctx, clientset, nodeInformer, "", stopChannel)
+	nodes, err := nodeutil.ReadyNodes(ctx, clientset, nodeInformer, "", nil)
 	if err != nil {
 		klog.Fatalf("%v", err)
 	}
 
-	lowNodeUtilizationStrategy := deschedulerapi.DeschedulerStrategy{
-		Enabled: true,
-		Params: &deschedulerapi.StrategyParameters{
-			NodeResourceUtilizationThresholds: &deschedulerapi.NodeResourceUtilizationThresholds{
-				Thresholds:       thresholds,
-				TargetThresholds: targetThresholds,
+	strategies.LowNodeUtilization(
+		ctx,
+		clientset,
+		deschedulerapi.DeschedulerStrategy{
+			Enabled: true,
+			Params: &deschedulerapi.StrategyParameters{
+				NodeResourceUtilizationThresholds: &deschedulerapi.NodeResourceUtilizationThresholds{
+					Thresholds: deschedulerapi.ResourceThresholds{
+						v1.ResourceMemory: 20,
+						v1.ResourcePods:   20,
+						v1.ResourceCPU:    85,
+					},
+					TargetThresholds: deschedulerapi.ResourceThresholds{
+						v1.ResourceMemory: 20,
+						v1.ResourcePods:   20,
+						v1.ResourceCPU:    90,
+					},
+				},
 			},
 		},
-	}
-
-	podEvictor := evictions.NewPodEvictor(
-		clientset,
-		evictionPolicyGroupVersion,
-		false,
-		0,
 		nodes,
-		false,
+		podEvictor,
 	)
 
-	strategies.LowNodeUtilization(ctx, clientset, lowNodeUtilizationStrategy, nodes, podEvictor)
 	time.Sleep(10 * time.Second)
 }
 
-func TestE2E(t *testing.T) {
-	// If we have reached here, it means cluster would have been already setup and the kubeconfig file should
-	// be in /tmp directory as admin.conf.
+func TestLowNodeUtilization(t *testing.T) {
 	ctx := context.Background()
+
 	clientSet, err := client.CreateClient(os.Getenv("KUBECONFIG"))
 	if err != nil {
 		t.Errorf("Error during client creation with %v", err)
 	}
 
+	stopChannel := make(chan struct{}, 0)
+
+	sharedInformerFactory := informers.NewSharedInformerFactory(clientSet, 0)
+	sharedInformerFactory.Start(stopChannel)
+	sharedInformerFactory.WaitForCacheSync(stopChannel)
+	defer close(stopChannel)
+
+	nodeInformer := sharedInformerFactory.Core().V1().Nodes()
+
 	nodeList, err := clientSet.CoreV1().Nodes().List(ctx, metav1.ListOptions{})
 	if err != nil {
 		t.Errorf("Error listing node with %v", err)
 	}
+
 	var nodes []*v1.Node
 	for i := range nodeList.Items {
 		node := nodeList.Items[i]
 		nodes = append(nodes, &node)
 	}
-	sharedInformerFactory := informers.NewSharedInformerFactory(clientSet, 0)
-	nodeInformer := sharedInformerFactory.Core().V1().Nodes()
+
+	testNamespace := &v1.Namespace{ObjectMeta: metav1.ObjectMeta{Name: "e2e-" + strings.ToLower(t.Name())}}
+	if _, err := clientSet.CoreV1().Namespaces().Create(ctx, testNamespace, metav1.CreateOptions{}); err != nil {
+		t.Fatalf("Unable to create ns %v", testNamespace.Name)
+	}
+	defer clientSet.CoreV1().Namespaces().Delete(ctx, testNamespace.Name, metav1.DeleteOptions{})
```

Comment:
I think it's better to check namespace too given we have checked if pods/rc are successfully deleted.

Reply:
Failing to delete a namespace is not a blocker for running other tests. So right now it's the best-effort.
On the other hand keeping a RC not deleted might influence tests where strategies like `LowNodeUtilization` are ran. So the test has to fail in case a RC is not deleted.